server/plugin/api_test.go

```diff
@@ -960,3 +961,172 @@ func TestGetUserChannelsForTeam(t *testing.T) {
 		})
 	}
 }
+
+func TestHandlePipelineApproveOrRejectRunRequest(t *testing.T) {
+	defer monkey.UnpatchAll()
+	mockAPI := &plugintest.API{}
+	mockCtrl := gomock.NewController(t)
+	mockedClient := mocks.NewMockClient(mockCtrl)
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
+	for _, testCase := range []struct {
+		description                            string
+		body                                   string
+		statusCode                             int
+		updatePipelineRunApprovalPostError     error
+		updatePipelineRunApprovalRequestError  error
+		getRunApprovalDetailsError             error
+		updatePipelineRunApprovalRequestStatus int
+		getRunApprovalDetailsStatus            int
+		isPayloadInvalid                       bool
+	}{
+		{
+			description: "HandlePipelineApproveOrRejectRunRequest: valid",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": "mockApproverID",
+				  "organization": "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineRunApprovalRequestStatus: http.StatusOK,
+			statusCode:                             http.StatusOK,
+			getRunApprovalDetailsStatus:            http.StatusOK,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectRunRequest: approved/rejected the request successfully but failed to update post",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": "mockApproverID",
+				  "organization": "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineRunApprovalRequestStatus: http.StatusOK,
+			updatePipelineRunApprovalPostError:     errors.New("approved/rejected the request successfully but failed to update post"),
+			statusCode:                             http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectRunRequest: failed to approve/reject request",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": "mockApproverID",
+				  "organization": "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineRunApprovalRequestStatus: http.StatusInternalServerError,
+			updatePipelineRunApprovalRequestError:  errors.New("not permitted to complete approval"),
```

Comment:
```suggestion
			updatePipelineRunApprovalRequestError:  errors.New("not permitted to complete the approval"),
```

Reply:
We cannot change this error. It is returned by Azure API